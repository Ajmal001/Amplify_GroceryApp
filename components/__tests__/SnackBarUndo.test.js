import React from 'react';
import renderer from 'react-test-renderer';
import SnackbarUndo from '../SnackBarUndo';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

test('renders correctly', () => {
  const tree = renderer.create(<SnackbarUndo
    visible={true}
    onDismissSnackBar={(x)=>x}
    undoAction={(x)=>x}
    snackContent={''} 
    />).toJSON();
  expect(tree).toMatchSnapshot();
});