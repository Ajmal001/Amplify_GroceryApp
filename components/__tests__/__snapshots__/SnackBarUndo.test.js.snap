// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityLiveRegion="polite"
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "borderRadius": 4,
          "elevation": 6,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 8,
          "opacity": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 6,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "flex": 1,
                "flexWrap": "wrap",
                "marginLeft": 16,
                "marginVertical": 14,
              },
              Object {
                "color": "#ffffff",
                "marginRight": 0,
              },
            ],
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "marginHorizontal": 8,
            "marginVertical": 6,
            "minWidth": "auto",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffca18",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Undo
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
